frontend/src/components/applyForm/widgets/RadioWidget.tsx

```diff
@@ -1,24 +1,70 @@
+"use client";
+
 import {
   enumOptionsIsSelected,
-  enumOptionsValueForIndex,
+  EnumOptionsType,
   FormContextType,
   optionId,
   RJSFSchema,
   StrictRJSFSchema,
 } from "@rjsf/utils";
 
-import { FocusEvent, useCallback, useMemo } from "react";
+import React, { FocusEvent, useCallback, useMemo } from "react";
 import { ErrorMessage, FormGroup, Radio } from "@trussworks/react-uswds";
 
 import { TextTypes, UswdsWidgetProps } from "src/components/applyForm/types";
 import { DynamicFieldLabel } from "./DynamicFieldLabel";
 import { getLabelTypeFromOptions } from "./getLabelTypeFromOptions";
 
-/** The `RadioWidget` is a widget for rendering a radio group.
- *  It is typically used with a string property constrained with enum options.
+/**
+ * The portion of `options` we care about here, with safe typing.
+ * We shape our data since RJSF’s options are loosely typed.
+ */
+type LocalOptions<S extends StrictRJSFSchema> = {
+  enumDisabled?: Array<string | number | boolean>;
+  emptyValue?: unknown;
+  enumOptions?: EnumOptionsType<S>[];
+  "widget-label"?: unknown;
+};
+
+/**
+ * normalizeForCompare
+ *
+ * Given an option’s value and the current field value, we force the current value type
+ * so it compares correctly with the option's representation.
+ */
+function normalizeForCompare(optionValue: unknown, current: unknown): unknown {
+  const optionIsBoolString = optionValue === "true" || optionValue === "false";
+  if (optionIsBoolString) {
+    if (current === true) return "true";
+    if (current === false) return "false";
+    if (current === "true" || current === "false") return current;
+    return undefined;
+  }
+  return current;
+}
+
+/**
+ * parseFromInputValue
  *
- * @param props - The `WidgetProps` for this component
+ * Convert an HTML input value (always a string) to either a boolean (for
+ * boolean radio groups) or the original enum value if we can find it.
  */
+function coerceFromString<S extends StrictRJSFSchema>(
+  raw: string,
+  enumOptions: ReadonlyArray<EnumOptionsType<S>>,
+): unknown {
+  const usesBoolStrings = enumOptions.some(
```

Comment:
[food for thought] I wonder if this function could be split out and used elsewhere, like in `normalizeForCompare` as well. I can envision something like

```
  const booleanStringToBoolean = (value: unknown): unknown => {
    if (value === "true" || value === "false") {
      return value.toString();
    }
    return value;
  }
```